staging/src/k8s.io/legacy-cloud-providers/azure/azure_routes.go

```diff
@@ -297,16 +297,18 @@ func (az *Cloud) CreateRoute(ctx context.Context, clusterName string, nameHint s
 		return nil
 	}
 
-	if !az.ipv6DualStackEnabled {
+	CIDRv6 := utilnet.IsIPv6CIDRString(string(kubeRoute.DestinationCIDR))
+	// if not dual-stack and is single stack IPv4, then get the IP of the
+	// machine from primary NIC.
+	if !az.ipv6DualStackEnabled && !CIDRv6 {
```

Comment:
I am not sure i understand the logic here. can you expand in the comments?

Reply:
Updated the comment. 

1. For single stack IPv4, we pick the primary ip config because the IPv4 IP is primary. 
2. For single stack IPv6, since we support on dual stack hosts, we get all the private IP address and then find the first one that's IPv6 because IPv6 private IP is always secondary.  
3. For dual stack, we get all the private IP address and then find the first one that matches the CIDR family.